presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/db/DbSourceExactMatchSelector.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.resourceGroups.db;
+
+import com.facebook.presto.spi.resourceGroups.ResourceGroupId;
+import com.facebook.presto.spi.resourceGroups.ResourceGroupSelector;
+import com.facebook.presto.spi.resourceGroups.SelectionContext;
+import io.airlift.json.JsonCodec;
+import io.airlift.log.Logger;
+import org.skife.jdbi.v2.exceptions.DBIException;
+
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static io.airlift.units.Duration.nanosSince;
+import static java.util.Objects.requireNonNull;
+
+public class DbSourceExactMatchSelector
+        implements ResourceGroupSelector
+{
+    private static final Logger log = Logger.get(DbSourceExactMatchSelector.class);
+    private static final JsonCodec<ResourceGroupId> resourceGroupIdCodec = JsonCodec.jsonCodec(ResourceGroupId.class);
+    private final ResourceGroupsDao dao;
+    private final String environment;
+    private final AtomicReference<Long> daoOfflineStart = new AtomicReference<>();
```

Comment:
Please give a more descriptive name to `daoOfflineStart`. This looks like the start timestamp for selector fetch failures, so how about sth like `fetchFailureStartTime`?

Reply:
@electrum suggested the original name. I'll change to whatever you guys agree on.

Reply:
if this is result of some suggestion please keep it as is then.